CGPasses.h.inc No such file (15.0.0 flang)

Looks the same as #54128

Centos7.9, gcc 11.3.0

```shell
cmake                                                 \    
  -DCMAKE_C_COMPILER=${HOST_GCC}/bin/gcc                                         \
  -DCMAKE_CXX_COMPILER=${HOST_GCC}/bin/g++                                       \
  -DCMAKE_BUILD_TYPE="Release"                                                   \
  -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}                                       \
  -DLLVM_TARGETS_TO_BUILD="X86"                                                  \
  -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;cross-project-tests;flang;libclc;lld;lldb;mlir;openmp;polly;pstl" \
  -DLLVM_ENABLE_RUNTIMES="compiler-rt;libc;libcxx;libcxxabi;libunwind" \
  -DGCC_INSTALL_PREFIX=${HOST_GCC}                                               \
  -DCMAKE_CXX_LINK_FLAGS="-L${HOST_GCC}/lib64 -Wl,-rpath,${HOST_GCC}/lib64"      \
  ../llvm
```
Terminates with error:
```console
.
.
.
[ 89%] Building CXX object tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/FrontendActions.cpp.o
In file included from /scr/llvm-project-15.0.0.src/flang/include/flang/Tools/CLOptions.inc:16,
                 from /src/llvm-project-15.0.0.src/flang/lib/Frontend/FrontendActions.cpp:480:
/scr/llvm-project-15.0.0.src/flang/include/flang/Optimizer/CodeGen/CodeGen.h:72:10: fatal error: flang/Optimizer/CodeGen/CGPasses.h.inc: No such file or directory
   72 | #include "flang/Optimizer/CodeGen/CGPasses.h.inc"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/FrontendActions.cpp.o] Error 1
gmake[1]: *** [tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/all] Error 2
gmake: *** [all] Error 2
```
I see (but don't really understand) in `/scr//llvm-project-15.0.0.src/flang/include/flang/Optimizer/CodeGen/CMakeLists.txt`:
```cmake
set(LLVM_TARGET_DEFINITIONS CGPasses.td)
mlir_tablegen(CGPasses.h.inc -gen-pass-decls -name OptCodeGen)
add_public_tablegen_target(FIROptCodeGenPassIncGen)
```

which looks to result in `/scr/llvm-project-15.0.0.src/build/tablegen_compile_commands.yml`:
```console
--- !FileInfo:
  filepath: "/scr/llvm-project-15.0.0.src/flang/include/flang/Optimizer/CodeGen/CGPasses.td"
  includes: "/scr/llvm-project-15.0.0.src/flang/include/flang/Optimizer/CodeGen;/scr/llvm-project-15.0.0.src/flang/include;/scr/llvm-project-15.0.0.src/build/tools/flang/include;/scr/llvm-project-15.0.0.src/build/include;/scr/llvm-project-15.0.0.src/llvm/include;/scr/llvm-project-15.0.0.src/llvm/../mlir/include;/scr/llvm-project-15.0.0.src/build/tools/mlir/include;/scr/llvm-project-15.0.0.src/build/tools/clang/include;/scr/llvm-project-15.0.0.src/llvm/../clang/include"
```
but beyone that I lose the trail...

I didn't have this problem with 13.0.1 nor 14.0.5

Same configuration (CentOS 7.9, gcc 11.3.0).
Still busted 15.0.6.

```
[ 89%] Building CXX object tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/FrontendActions.cpp.o
In file included from /scr/llvm-15.0.6/llvm-project-15.0.6.src/flang/include/flang/Tools/CLOptions.inc:16,
                 from /scr/llvm-15.0.6/llvm-project-15.0.6.src/flang/lib/Frontend/FrontendActions.cpp:480:
/scr/llvm-15.0.6/llvm-project-15.0.6.src/flang/include/flang/Optimizer/CodeGen/CodeGen.h:72:10: fatal error: flang/Optimizer/CodeGen/CGPasses.h.inc: No such file or directory
   72 | #include "flang/Optimizer/CodeGen/CGPasses.h.inc"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/FrontendActions.cpp.o] Error 1
gmake[1]: *** [tools/flang/lib/Frontend/CMakeFiles/obj.flangFrontend.dir/all] Error 2
gmake: *** [all] Error 2
```

--- a/flang/lib/Frontend/CMakeLists.txt
+++ b/flang/lib/Frontend/CMakeLists.txt
@@ -15,6 +15,7 @@ add_flang_library(flangFrontend
 
   DEPENDS
   FIRDialect
+  FIROptCodeGenPassIncGen
   FIROptTransformsPassIncGen
   HLFIRDialect
   MLIRIR

